### PR TITLE
feat(app): add Worldcoin stamp

### PR DESCRIPTION
**Pull request by 0xflywill**
adds the Worldcoin stamp which proves someone has been anonymously verified as a unique human with World ID (docs at https://id.worldcoin.org)

### CODE REVIEW

File: app/components/PlatformCards/WorldIDPlatform.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  const { handleAddStamps, allProvidersState } = useContext(CeramicContext);
+  const { address, signer } = useContext(UserContext);
+  const [isLoading, setLoading] = useState(false);
+  const [canSubmit, setCanSubmit] = useState(false);
+
+  const toast = useToast();
+
+  // find all providerIds
+  const providerIds =
+    STAMP_PROVIDERS["Brightid"]?.reduce((all, stamp) => {
```
Comment (m1guelpf) [inline app/components/PlatformCards/WorldIDPlatform.tsx:57]: Should this be WorldID instead?

File: iam/src/providers/worldid.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+
+// ----- Credential verification
+import { getAddress } from "../utils/signer";
+import axios from "axios";
+
+// API endpoint to verify World ID ZKP
+const VERIFY_ENDPOINT = "https://developer.worldcoin.org/api/v1/verify";
+// Static action ID to verify ZKP (should never change in prod! for local testing, new IDs can be created in developer.worldcoin.org)
+export const WORLD_ID_ACTION_ID = "wid_5047fd9af3d4a665da9a44251270d6b2";
```
Comment (m1guelpf) [inline iam/src/providers/worldid.ts:12]: Let's move this to an environment variable maybe?

**Comment by akathm** [comment]
Hey @0xflywill 

I've asked for this from some engineers on the Worldcoin team before, but if you all would be open to providing 3rd party audits of the camera itself as well as the biometric data ingestion, this would help us to perform a risk assessment on this stamp.

Until the time that this is open source code and/or has been verified by a neutral 3rd party, however, we cannot confirm whether this stamp poses a direct threat to users of Gitcoin Passport.

**Comment by 0xflywill** [comment]
Hey @akathm

Thanks for the update! In general the World ID Protocol, Orb device, and more are now open source at http://github.com/worldcoin. Multiple audits across the stack have also been performed and are in the process of being published.

A few resources you might find useful:

- Open Sourcing the Orb: https://worldcoin.org/blog/engineering/opening-orb-look-inside-worldcoin-biometric-imaging-device
- Privacy Technical Deep Dive: https://worldcoin.org/blog/developers/privacy-deep-dive
- Decentralizing Worldcoin: https://worldcoin.org/blog/worldcoin/primer-on-decentralization-at-worldcoin

In any case, closing out this specific PR as when the integration was developed last year it used the now deprecated World ID Alpha. Feel free to let us know when the team has bandwidth to review additional Stamps, and we’ll be happy to support the community in building an updated version using the new World ID Beta.

**Pull request closed without merging**